frontend/src/api/k8s/templates.ts

```diff
@@ -101,3 +111,44 @@ export const createServingRuntimeTemplate = (
     return Promise.reject(e);
   }
 };
+
+export const updateServingRuntimeTemplate = (
+  name: string,
+  body: string,
+  namespace: string,
+): Promise<TemplateKind> => {
+  try {
+    const template = assembleTemplate(body, namespace);
+    return dryRunServingRuntimeForTemplateCreation(template, namespace).then(() =>
+      k8sPatchResource({
+        model: TemplateModel,
+        queryOptions: { name, ns: namespace },
+        patches: [
+          {
+            op: 'replace',
+            path: '/metadata/annotations/openshift.io~1description',
+            value: getDescriptionFromK8sResource(template),
+          },
+          {
+            op: 'replace',
+            path: '/metadata/annotations/openshift.io~1display-name',
```

Comment:
We don't check display name for the templates so there could be the rows showing the same name if they have different metadata name and same display name. Let me know your thoughts @lucferbux @vconzola
<img width="1509" alt="Screenshot 2023-04-26 at 9 28 41 PM" src="https://user-images.githubusercontent.com/37624318/234590536-63f62d48-e8ba-47a3-8f89-112d3e5ef0ad.png">

Reply:
If possible, we should detect this condition and alert the the user - e.g., "A runtime with this name already exists. To avoid duplicate names appearing in the runtimes table, change the name in the <display_name> field in the YAML file." Action options on the alert modal would be "Ignore and save" and "Cancel". We might also consider adding a "Last saved" column to the table so the user has some way to distinguish between runtimes withe same display name.

Reply:
cc @kywalker-rh we have no functionality like this -- is this desired throughout?

@vconzola would you be okay not having that first pass? I talk to Kyle A LOT about consistency in the Dashboard. Doing really special and desirable things in one spot sorta softens the holistic nature of the Dashboard. We should aim to support the same functionality _everywhere_ in the dashboard as a single request so the user experience remains consistent and predictable -- imo.

Reply:
Yeah, I'm ok with letting this go for a first pass. But we should add the Last saved column in that case. How are we handling dupes in other parts of the UI - just letting them be? Btw, I would never advocate for consistency over better UX.

Reply:
Also, not sure what you mean we don't have this feature elsewhere in the UI. I just tried creating duplicate projects with the same name and they get flagged. This situation is unique, however, because the display name is not user generated. It's taken directly from the YAML file, per Chris. (At least I think it was Chris.) If we let the user name their own runtimes we could be consistent with what we're already doing and not allow duplicate names.

Reply:
I responded [on slack](https://redhat-internal.slack.com/archives/C043A9D5DMW/p1682543093950959) -- this is a process concern, not me blocking UX. See my reply there for my reasoning.

Reply:
Just to update the information here:

1. This same situation happens with other resources with display name, if we decide to change it, I would say we track it for the future and do a global refactor.
2. The main issue here lies that [templates](https://docs.openshift.com/container-platform/4.9/openshift_images/using-templates.html) can be misleading if we abstract the user from them, the user is actually filling the spec for a **ServingRuntime** which will be stored in a template, this template has it's own metadata outside the ServingRuntime spec, so the first question would be if we either **update the metadata from the template** or just **treat the serving runtime metadata** as the single source of truth.

Reply:
Ok, based on conversations, my take on this would be this one:

1. Leave the Template name alone (maybe random names)
2. Use the Serving Runtime `name` and `display-name` as source of truth
3. When editing, make the Serving Runtime `name` immutable and fail if we intend to modify it

I think that's what I got from the latest conversations, let me know if I misunderstood/missed something @andrewballantyne @DaoDaoNoCode